src/main/scala/BintrayPlugin.scala

```diff
@@ -0,0 +1,194 @@
+      else Bintray.defaultMavenRepository
+    },
+    publishMavenStyle := {
+      if (sbtPlugin.value) false else publishMavenStyle.value
+    },
+    bintrayPackageLabels := Nil,
+    description in bintray <<= description,
+    // note: publishTo may not have dependencies. therefore, we can not rely well on inline overrides
+    // for inline credentials resolution we recommend defining bintrayCredentials _before_ mixing in the defaults
+    // perhaps we should try overriding something in the publishConfig setting -- https://github.com/sbt/sbt-pgp/blob/master/pgp-plugin/src/main/scala/com/typesafe/sbt/pgp/PgpSettings.scala#L124-L131
+    publishTo in bintray <<= publishToBintray,
+    resolvers in bintray := {
+      Bintray.buildResolvers(bintrayCredentialsFile.value,
+        bintrayOrganization.value,
+        bintrayRepository.value)
+    },
+    credentials in bintray := {
+      Credentials(bintrayCredentialsFile.value) :: Nil
+    },
+    bintrayPackageAttributes := {
+      if (sbtPlugin.value) Map(AttrNames.sbtPlugin -> Seq(Attr.Boolean(sbtPlugin.value)))
+      else Map.empty
+    },
+    bintrayVersionAttributes := {
+      val scalaVersions = crossScalaVersions.value
+      val sv = Map(AttrNames.scalas -> scalaVersions.map(Attr.Version(_)))
+      if (sbtPlugin.value) sv ++ Map(AttrNames.sbtVersion-> Seq(Attr.Version(sbtVersion.value)))
+      else sv
+    },
+    bintrayOmitLicense := {
+      if (sbtPlugin.value) sbtPlugin.value
+      else false
+    },
+    bintrayEnsureLicenses := {
+      Bintray.ensureLicenses(licenses.value, bintrayOmitLicense.value)
+    },
+    bintrayEnsureCredentials := {
+      Bintray.ensuredCredentials(bintrayCredentialsFile.value).get
+    },
+    bintrayEnsureBintrayPackageExists <<= ensurePackageTask,
+    bintrayUnpublish := {
+      val e1 = bintrayEnsureBintrayPackageExists
+      val e2 = bintrayEnsureLicenses
+      val repo = bintrayRepo.value
+      repo.unpublish(bintrayPackage.value, version.value, sLog.value)
+    },
+    bintrayRemoteSign := {
+      val repo = bintrayRepo.value
+      repo.remoteSign(bintrayPackage.value, version.value, sLog.value)
+    },
+    bintraySyncMavenCentral := {
+      val repo = bintrayRepo.value
+      repo.syncMavenCentral(bintrayPackage.value, version.value, credentials.value, sLog.value)
+    },
+    bintrayRelease := {
+      val _ = publishVersionAttributesTask.value
+      val repo = bintrayRepo.value
+      repo.release(bintrayPackage.value, version.value, sLog.value)
+    }
+  ) ++ Seq(
+    resolvers <++= resolvers in bintray,
+    credentials <++= credentials in bintray,
+    publishTo <<= publishTo in bintray, 
+    publish := dynamicallyPublish.value
+  )
+
+  private def vcsUrlTask: Initialize[Task[Option[String]]] =
+    task {
+      Bintray.resolveVcsUrl.recover { case _ => None }.get
+    } tag(Git)
+
+  private def dynamicallyPublish: Initialize[Task[Unit]] =
+    taskDyn {
+      val _ = publish.value
```

Comment:
If I don't assign `publish.value` to anything, then I think the compiler warns me saying "hey, you have an expression that's not doing anything." So I have to assign it to something, but it's not like I need it later, so the convention is to use `_` for that purpose.

Not sure if it has any magical power, since I don't think you can use more than one of them in scope. But I guess it's sort of analogous to using pattern on the lhs of value definition, and using `_` as a pattern.